serializer/serializer.go

```diff
@@ -368,16 +377,12 @@ func (s *Serializer) WritePayload(payload Serializable, deSeriMode DeSerializati
 }
 
 // WriteString writes the given string to the Serializer.
-func (s *Serializer) WriteString(str string, errProducer ErrProducer) *Serializer {
+func (s *Serializer) WriteString(str string, lenType SeriSliceLengthType, errProducer ErrProducer) *Serializer {
 	if s.err != nil {
 		return s
 	}
 
-	if err := binary.Write(&s.buf, binary.LittleEndian, uint16(len(str))); err != nil {
-		s.err = errProducer(err)
-		return s
-	}
-
+	_ = s.writeSliceLength(len(str), lenType, errProducer)
```

Comment:
ditto the same as above, check whether an error was produced with `s.err != nil ...`.